bokehjs/src/lib/models/comparisons/nan_sorter.ts

```diff
@@ -0,0 +1,36 @@
+import {Comparison} from "./comparison"
+import type * as p from "core/properties"
+
+export namespace NanSorter {
+  export type Attrs = p.AttrsOf<Props>
+
+  export type Props = Comparison.Props & {
+    ascending_first: p.Property<boolean>
+  }
+}
+
+export interface NanSorter extends NanSorter.Attrs {}
+
+export class NanSorter extends Comparison {
+  declare properties: NanSorter.Props
+
+  constructor(attrs?: Partial<NanSorter.Attrs>) {
+    super(attrs)
+  }
+
+  static {
+    this.define<NanSorter.Props>(({Bool}) => ({
+      ascending_first: [ Bool, false ],
+    }))
+  }
+
+  protected compute(x: any, y: any): number {
+    if (isNaN(x)) {
+      return this.ascending_first ? -1 : 1
+    }
+    if (isNaN(y)) {
+      return this.ascending_first ? 1 : -1
+    }
+    return x==y ? 0 : x < y ? -1 : 1
+  }
```

Comment:
`isNaN()` can somewhat work in place of `isNumber()`, but it's finicky, and only as long as `any` type is involved on the type-level. Given the inputs to this method can be anything, I would change the types to `unknown` and add `isNumber()` guards to make sure `isNaN()` has actual numbers to work with.

Reply:
I made this change but it also limited the comparison in general to numeric types, I could have imagined sorting nans first or last relative to non-numeric types in case someone used nan as "missing". Currently returning 0 for non-numeric types, but perhaps an error would be better?